great_expectations/expectations/core/expect_query_results_to_match_source.py

```diff
@@ -275,5 +292,178 @@ def _validate(
                 "result": {
                     "unexpected_count": unexpected_count,
                     "unexpected_percent": unexpected_percent,
+                    "details": {
+                        "missing_rows": missing_rows,
+                        "unexpected_rows": unexpected_rows,
+                    },
                 },
             }
+
+    @override
+    @classmethod
+    @renderer(renderer_type=AtomicDiagnosticRendererType.OBSERVED_VALUE)
+    def _atomic_diagnostic_observed_value(
+        cls,
+        configuration: Optional[ExpectationConfiguration] = None,
+        result: Optional[ExpectationValidationResult] = None,
+        runtime_configuration: Optional[dict] = None,
+    ) -> list[RenderedAtomicContent]:
+        details = cls._get_details_from_results(result)
+
+        missing_rows: list[dict[str, Any]] = details["missing_rows"]
+        unexpected_rows: list[dict[str, Any]] = details["unexpected_rows"]
+        missing_rows_table = cls._create_observed_values_table(missing_rows)
+        unexpected_rows_table = cls._create_observed_values_table(unexpected_rows)
+
+        return [
+            cls._create_table_rendered_atomic_content(
+                missing_rows_table,
+                no_table_msg="No missing rows",
+            ),
+            cls._create_table_rendered_atomic_content(
+                unexpected_rows_table,
+                no_table_msg="No unexpected rows",
+            ),
+        ]
+
+    @classmethod
+    def _get_details_from_results(
+        cls, result: Optional[ExpectationValidationResult]
+    ) -> dict[str, Any]:
+        assert result and result.result, "Must have result"
+        details = result.result.get("details")
+        assert isinstance(details, dict), "Details must exist and be a dict"
+        return details
+
+    @classmethod
+    def _create_table_rendered_atomic_content(
+        cls,
+        table: list[list[RendererTableValue]],
+        no_table_msg: str,
+    ) -> RenderedAtomicContent:
+        if not table:
+            return RenderedAtomicContent(
+                name=AtomicDiagnosticRendererType.OBSERVED_VALUE,
+                value=RenderedAtomicValue(
+                    schema={"type": "StringValueType"},
+                    template=no_table_msg,
+                ),
+                value_type="StringValueType",
+            )
+
+        return RenderedAtomicContent(
+            name=AtomicDiagnosticRendererType.OBSERVED_VALUE,
+            value=RenderedAtomicValue(
+                schema={"type": "TableType"},
+                header_row=table[0],
+                table=table[1:],
+            ),
+            value_type="TableType",
+        )
```

Comment:
I think we want to include a template and params here that show the missing/unexpected row count.